### PR TITLE
feat: immutable options in random.alpha methods

### PR DESCRIPTION
**Pull request by ST-DDT**
Similar to other methods I changed `random.alpha()` and `random.alphaNumeric()` to don't alter their options.

Hints for reviewers:
- I moved the default values to the method signature, to improve the api doc generation.
- I also removed the array length check, as the called method does nothing, if the array is empty, and does so almost NO-OP.
  I can re-add the check, if you insist.

### CODE REVIEW

File: src/random.ts [hunk 1]
```diff
@@ -400,28 +400,22 @@ export class Random {
    */
   // TODO @Shinigami92 2022-02-14: Tests covered `(count, options)`, but they were never typed like that
   alpha(
-    options?:
+    options:
```
[the remark this answers is not in the document]
Comment (ST-DDT) [reply src/random.ts:403]: Did you miss the default value?
https://github.com/faker-js/faker/blob/c8ae059dcdd7325e930fdd3d63b7217c806cd2c2/src/random.ts#L405-L409

[the remark this answers is not in the document]
Comment (pkuczynski) [reply src/random.ts:403]: The default values would not work if user will pass partial `options` object, right?

[the remark this answers is not in the document]
Comment (Shinigami92) [reply src/random.ts:403]: Oh my god, seems it's horrible to read 👀 
Sry for false alarm

[the remark this answers is not in the document]
Comment (Shinigami92) [reply src/random.ts:403]: > The default values would not work if user will pass partial `options` object, right?

Yes 🙁 , but maybe line 416-418 solves this?

[the remark this answers is not in the document]
Comment (pkuczynski) [reply src/random.ts:403]: > > The default values would not work if user will pass partial `options` object, right?
> 
> 
> 
> Yes 🙁 , but maybe line 416-418 solves this?

It does, but then the same logic is in two places... :/

[the remark this answers is not in the document]
Comment (ST-DDT) [reply src/random.ts:403]: Actually 4:
- jsdocs
- signature
- impl
- test(s)

If js would support better syntax, then we would only need 2 (+ jsdocs can be cross-validated).

File: src/random.ts [hunk 2]
```diff
@@ -477,11 +470,9 @@ export class Random {
    */
   alphaNumeric(
     count: number = 1,
-    options: { bannedChars?: string[] } = {}
+    options: { bannedChars?: string[] } = { bannedChars: [] }
   ): string {
-    if (options.bannedChars == null) {
-      options.bannedChars = [];
-    }
+    const bannedChars = options.bannedChars ?? [];
```
[the remark this answers is not in the document]
Comment (ST-DDT) [reply src/random.ts:475]: > In packages like VueUse we are using a more advanced syntax like

Lets discuss this later today.

> Maybe we should use `options: { bannedChars?: readonly string[] }` in line 473, as far as I know, this also allows readonly and non-readonly arrays to be passed.

Will do so.

**Review by Shinigami92 — CHANGES_REQUESTED** [review]
Previously there was no failing test, please add one to test that the options are not mutated

**Review by Shinigami92 — DISMISSED** [review]

File: src/random.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (pkuczynski) [reply src/random.ts:403]: I believe @Shinigami92 implemented this differently in his PR keeping the logic only in the implementation, and I must say I like it more... regardless of my opinion, we should do this consistently. Now we don't, so for me its a blocker...

[the remark this answers is not in the document]
Comment (ST-DDT) [reply src/random.ts:403]: I changed it please review again.

**Review by pkuczynski — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**